packages/medusa-payment-stripe/src/api/routes/hooks/stripe.js

```diff
@@ -39,9 +41,12 @@ export default async (req, res) => {
       break
     case "payment_intent.amount_capturable_updated":
       if (!order) {
-        await cartService.setPaymentSession(cartId, "stripe")
-        await cartService.authorizePayment(cartId)
-        await orderService.createFromCart(cartId)
+        await manager.transaction(async (manager) => {
+          const cartServiceTx = cartService.withTransaction(manager)
+          await cartServiceTx.setPaymentSession(cartId, "stripe")
+          await cartServiceTx.authorizePayment(cartId)
+          await orderService.withTransaction(manager).createFromCart(cartId)
```

Comment:
what do you mean? to call the completion strategy instead and use the idempotency?

Reply:
Yes, my thinking was, that we could call the cart completion strategy here as we are essentially performing the same steps. Would reduce the amount of code to maintain.

```
const completionStrat: AbstractCartCompletionStrategy = req.scope.resolve("cartCompletionStrategy")
...
case "payment_intent.amount_capturable_updated":
      if (!order) {
        await cartService.setPaymentSession(cartId, "stripe")
        const { response_code, response_body } = await completionStrat.complete(
          cartId,
          idempotencyKey,
          req.request_context
        )
      }
      break
```

I would have to run some tests to ensure we can use it, but it would be quite nice.

Reply:
I think that it would be possible, specially that the actual impl does not take into account the tax line creation